src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationRow/integrationDirectoryApplicationRow.tsx

```diff
@@ -0,0 +1,186 @@
+import SentryTypes from 'app/sentryTypes';
+import {PanelItem} from 'app/components/panels';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import CircleIndicator from 'app/components/circleIndicator';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import {Organization, SentryApp, SentryAppInstallation} from 'app/types';
+import theme from 'app/utils/theme';
+
+const INSTALLED = 'Installed';
+const NOT_INSTALLED = 'Not Installed';
+const PENDING = 'Pending';
+
+type Props = {
+  app: SentryApp;
+  organization: Organization;
+  install?: SentryAppInstallation;
+  isOnIntegrationPage: boolean;
+  ['data-test-id']?: string;
+};
+
+export default class SentryApplicationRow extends React.PureComponent<Props> {
+  static propTypes = {
+    app: SentryTypes.SentryApplication,
+    organization: SentryTypes.Organization.isRequired,
+    install: PropTypes.object,
+    isOnIntegrationPage: PropTypes.bool,
+  };
+
+  get isInternal() {
+    return this.props.app.status === 'internal';
+  }
+
+  hideStatus() {
+    //no publishing for internal apps so hide the status on the developer settings page
+    return this.isInternal && !this.props.isOnIntegrationPage;
+  }
+
+  renderStatus() {
+    const {app, isOnIntegrationPage} = this.props;
+    const isInternal = this.isInternal;
+    const status = this.installationStatus;
+    if (this.hideStatus()) {
+      return null;
+    }
+    if (isOnIntegrationPage) {
+      //if internal and we show installation status, we don't show the learn more
+      if (isInternal) {
+        return <StatusIndicator status={status} isInternal={isInternal} />;
+      }
+      return (
+        <React.Fragment>
+          <StatusIndicator status={status} isInternal={false} />
+        </React.Fragment>
+      );
+    }
+    return <PublishStatus status={app.status} />;
+  }
+
+  get installationStatus() {
+    if (this.props.install) {
+      return capitalize(this.props.install.status);
+    }
+
+    return NOT_INSTALLED;
+  }
+
+  linkToEdit() {
+    const {isOnIntegrationPage} = this.props;
+    // show the link if the app is internal or we are on the developer settings page
+    // We don't want to show the link to edit on the main integrations list unless the app is internal
+    return this.isInternal || !isOnIntegrationPage;
+  }
+
+  render() {
+    const {app, organization} = this.props;
+    return (
+      <SentryAppItem data-test-id={app.slug}>
+        <StyledFlex>
+          <PluginIcon size={36} pluginId={app.slug} />
+          <SentryAppBox>
+            <SentryAppName hideStatus={this.hideStatus()}>
+              {this.linkToEdit() ? (
+                <SentryAppLink
+                  to={`/settings/${organization.slug}/developer-settings/${app.slug}/`}
+                >
+                  {app.name}
+                </SentryAppLink>
+              ) : (
+                <SentryAppLink
+                  to={`/settings/${organization.slug}/integrations/${app.slug}`}
```

Comment:
@NisanthanNanthakumar I realize the card doesn't explain this (the future one does, but we need it a different URL. It should be `/settings/${organization.slug}/sentry-apps/${app.slug}`.